### ISSUE TITLE
TypeScript compiler bug.

### ISSUE THREAD

**Issue by alexander-matsievsky** [Microsoft/TypeScript#3992]
Good day!

I've isolated the offending source to a separate [branch](https://github.com/alexander-matsievsky/circuitry.ts/blob/ts_compiler_error_2015-07-23T02colon51colon30dot461Z/examples/todo/classic/js/main.ts#L367-377). The commenting out of the specified lines resolves the error.

To run the code, clone the repo and compile, no `npm` or build required.

I'd have tried to fix this bug myself, but I'm not sure I have the competence required. I'll be glad to provide you with additional details upon request.

##### OS version

```
uname -a
Linux matsievskyav-Z87X-D3H 3.16.0-29-generic #39-Ubuntu SMP Tue Dec 16 20:54:13 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

##### Node.js version

```
~/.nvm/versions/node/v0.12.3/bin/node --version
v0.12.3
```

##### Compiler version

```
~/.nvm/versions/node/v0.12.3/node_modules/.bin/tsc --version
message TS6029: Version 1.5.0-beta
```

##### Error

```
~/.nvm/versions/node/v0.12.3/node_modules/.bin/tsc --target es5 --module commonjs ./examples/todo/classic/js/main.ts
/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:21566
                            !(operand.kind === 157 && node.parent.kind === 158
                                                                 ^
TypeError: Cannot read property 'kind' of undefined
    at emitParenExpression (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:21566:66)
    at emitBareNode (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:24092:32)
    at emitNodeWorker (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:23997:21)
    at emit (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:23964:17)
    at emitPropertyAccess (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:21425:17)
    at emitBareNode (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:24080:32)
    at emitNodeWorker (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:23997:21)
    at emit (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:23964:17)
    at emitAssignment (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:22093:21)
    at emitBindingElement (/home/matsievskyav/.nvm/versions/node/v0.12.3/node_modules/typescript/bin/tsc.js:22237:25)
---------------------------------------------------------------------------------------------------------------------
```

**Comment by alexander-matsievsky** [comment]
After some playing with the code I've found out that this compiles OK:

```ts
//.Effect<{id:UUID; phaze:Editing}>(DESCRIBE_LATER, (In, Out) => Out.editing, (Mem, editing) => {
    let li = <HTMLLIElement>Mem.DOM.todoList.querySelector(`[data-id="${editing.id}"]`),
        classList = li.classList;
//})
```

This produces the error:

```ts
//.Effect<{id:UUID; phaze:Editing}>(DESCRIBE_LATER, (In, Out) => Out.editing, (Mem, editing) => {
    let {classList} = <HTMLLIElement>Mem.DOM.todoList.querySelector(`[data-id="${editing.id}"]`);
//})
```

**Comment by RyanCavanaugh** [comment]
Verified repro at the specified tag. This no longer crashes in `master` or at `v1.5.3`, so I'd recommend updating to either the latest or the released 1.5 version. Thanks for the detailed report, though!

**Comment by alexander-matsievsky** [comment]
@RyanCavanaugh Oh, it's great to know! Thank you!
